stash_datacite/app/controllers/stash_datacite/publications_controller.rb

```diff
@@ -1,37 +1,51 @@
 require_dependency 'stash_datacite/application_controller'
 require 'httparty'
+require 'stash/import/cross_ref'
 
 module StashDatacite
   class PublicationsController < ApplicationController
     # include HTTParty
-
-    # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
     def update
       @se_id = StashEngine::Identifier.find(params[:internal_datum][:identifier_id])
-      @pub_issn = StashEngine::InternalDatum.find_or_create_by(stash_identifier: @se_id, data_type: 'publicationISSN')
-      @msid = StashEngine::InternalDatum.find_or_create_by(stash_identifier: @se_id, data_type: 'manuscriptNumber')
-      @doi = StashEngine::InternalDatum.find_or_create_by(stash_identifier: @se_id, data_type: 'publicationDOI')
-      issn = params[:internal_datum][:publication_issn]
-      msid = params[:internal_datum][:msid]
-      doi = params[:internal_datum][:doi]
-      @pub_issn.update(value: issn) unless issn.blank?
-      @msid.update(value: msid) unless msid.blank?
-      @doi.update(value: doi) unless doi.blank?
+      save_form_to_internal_data
       respond_to do |format|
         format.js do
           if params[:internal_datum][:do_import] == 'true'
             # take action to do the actual import and reload the page with javascript
-            update_metadata
-            render 'update' # just the standard update in the associated view directory
+            @resource = @se_id.latest_resource
+            if @doi.value.blank? && !@msid.value.blank?
+              update_manuscript_metadata
+            else
+              update_doi_metadata
```

Comment:
this may be fine, just wanted to point out that the `update_manuscript_metadata` would not be called if the user provided both a DOI and an MSID.
An alternative would be:
```ruby
update_manuscript_metadata unless @msid.value.blank?
update_doi_metadata unless @doi.value.blank?
```

Reply:
This is a good point, Brian.  I'll change this.